### PR TITLE
ecwolf: replace `fetchurl` with `fetchFromBitbucket` & 1.3.3 -> 1.4.0

### LINKED ISSUE #211388: ECWolf is out of date (1.3.3 -> 1.4.0)
###### Checklist

- [x] Checked the [nixpkgs master branch](https://github.com/NixOS/nixpkgs)

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

###### Project name
`nix search` name: ecwolf

current version: 1.3.3
desired version: 1.4.0

###### Notify maintainers

maintainers: @svanderburg

###### Note for maintainers

Please tag this issue in your PR.

### PR DESCRIPTION
**Pull request by Jayman2000**
###### Description of changes

This replaces `fetchurl` with `fetchFromBitbucket` to simplify `ecwolf`’s `default.nix` and updates `ecwolf` to 1.4.0. See the individual commit messages for details.

Closes #211388.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by SuperSandro2000 — COMMENTED** [review]
Please mark the package broken on darwin unless you can fix the compile error

**Comment by Jayman2000** [comment]
> Please mark the package broken on darwin unless you can fix the compile error

Done.

**Review by NickCao — APPROVED** [review]

File: pkgs/games/ecwolf/default.nix [hunk 1]
```diff
@@ -45,6 +61,7 @@ stdenv.mkDerivation rec {
   '';
 
   meta = with lib; {
+    broken = stdenv.isDarwin;
```
Comment (SuperSandro2000) [inline pkgs/games/ecwolf/default.nix:64]: broken usually goes last in meta. That it was first was a mistake from a treewide change. Also a short comment why is usually very helpful for people trying to fix it.

Comment (Jayman2000) [reply pkgs/games/ecwolf/default.nix:64]: I just submitted #212871 which addresses this issue.

File: pkgs/games/ecwolf/default.nix [hunk 2]
```diff
@@ -1,16 +1,32 @@
...
+, copyDesktopItems
+, cmake
+, pkg-config
+, zlib
+, bzip2
+, libjpeg
+, SDL2
+, SDL2_net
+, SDL2_mixer
+, gtk2
```
[the remark this answers is not in the document]
Comment (Jayman2000) [reply pkgs/games/ecwolf/default.nix:14]: I just submitted #212871 which addresses this issue.

**Pull request merged**